apps/user-profile/src/app/app.component.ts

```diff
@@ -33,11 +35,17 @@ export class AppComponent implements OnInit {
       return;
     }
     this.attributesManagerService.getUserAttributes(this.store.getPerunPrincipal().userId).subscribe(atts =>{
-     const prefLang = atts.find(elem => elem.friendlyName === 'preferredLanguage');
-      if(prefLang && prefLang.value) {
-        // @ts-ignore
-        this.translateService.use(prefLang.value);
-      }
+      const prefLang = atts.find(elem => elem.friendlyName === 'preferredLanguage');
+      this.route.queryParams.subscribe(params => {
+        const langs = this.store.get('supported_languages');
+        const urlLang = params['lang'];
+        if(urlLang && langs.includes(urlLang)) {
+          this.translateService.use(urlLang);
+        } else if(prefLang && prefLang.value) {
+          // @ts-ignore
+          this.translateService.use(prefLang.value);
+        }
+      });
```

Comment:
isnt this block of code repeating throught the application? if so store it in one place and call it from all the places where you need it